utils/swift_build_support/swift_build_support/products/swiftpm.py

```diff
@@ -2,16 +2,72 @@
 #
 # This source file is part of the Swift.org open source project
 #
-# Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+# Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
 # Licensed under Apache License v2.0 with Runtime Library Exception
 #
 # See https://swift.org/LICENSE.txt for license information
 # See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 #
 # ----------------------------------------------------------------------------
 
-from . import product
+import os
+import platform
 
+from . import product
+from .. import shell
+from .. import targets
 
 class SwiftPM(product.Product):
-    pass
+    @classmethod
+    def product_source_name(cls):
+        return "swiftpm"
+
+    @classmethod
+    def is_build_script_impl_product(cls):
+        return False
+
+    def build(self, host_target):
+        run_build_script_helper('build', host_target, self, self.args)
+
+    def test(self, host_target):
+        run_build_script_helper('test', host_target, self, self.args)
+
+    def install(self, host_target):
+        run_build_script_helper('install', host_target, self, self.args)
+
+def run_build_script_helper(action, host_target, product, args):
+    script_path = os.path.join(
+        product.source_dir, 'Utilities', 'bootstrap')
+    toolchain_path = args.install_destdir
+    if platform.system() == 'Darwin':
+        # The prefix is an absolute path, so concatenate without os.path.
+        toolchain_path += \
+            product.targets.darwin_toolchain_prefix(args.install_prefix)
```

Comment:
Not part of this PR but: We should factor this building of the toolchain path out to the driver and not copy it into every PR. I’ll take care of it

Reply:
yeah